src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/ProcessorClient.cs

```diff
@@ -32,50 +32,73 @@ internal class ProcessorClient(
         {
             ReleaseFileId = releaseFileId,
+            DataSetId = dataSetId
         };
 
-        var response = await httpClient
-            .PostAsJsonAsync("api/CreateDataSet", request, cancellationToken: cancellationToken);
+        return await SendPost<NextDataSetVersionCreateRequest, CreateDataSetResponseViewModel>(
+            "api/CreateNextDataSetVersion",
+            request,
+            cancellationToken);
+    }
 
-        if (!response.IsSuccessStatusCode)
-        {
-            switch (response.StatusCode)
-            {
-                case HttpStatusCode.BadRequest:
-                    return new BadRequestObjectResult(
-                        await response.Content
-                            .ReadFromJsonAsync<ValidationProblemViewModel>(cancellationToken: cancellationToken)
-                    );
-                default:
-                    var message = await response.Content.ReadAsStringAsync(cancellationToken);
-                    logger.LogError($"""
-                                     Failed to create data set with status code: {response.StatusCode}. Message:
-                                     {message}
-                                     """);
-                    response.EnsureSuccessStatusCode();
-                    break;
-            }
-        }
+    public async Task<Either<ActionResult, Unit>> DeleteDataSetVersion(
+        Guid dataSetVersionId,
+        CancellationToken cancellationToken = default)
+    {
+        return await SendDelete(
+            $"api/DeleteDataSetVersion/{dataSetVersionId}",
+            cancellationToken);
+    }
 
-        var content = await response.Content.ReadFromJsonAsync<CreateDataSetResponseViewModel>(
-            cancellationToken: cancellationToken
-        );
+    private async Task<Either<ActionResult, TResponse>> SendPost<TRequest, TResponse>(
+        string url,
+        TRequest request,
+        CancellationToken cancellationToken = default)
+        where TResponse : class
+    {
+        return await SendRequest<TResponse>(() =>
+                httpClient.PostAsJsonAsync(
+                    url,
+                    request,
+                    cancellationToken: cancellationToken),
+            cancellationToken);
+    }
 
-        return content
-            ?? throw new Exception("Could not deserialize the response from the Public Data Processor.");
+    private async Task<Either<ActionResult, Unit>> SendDelete(
+        string url,
+        CancellationToken cancellationToken = default)
+    {
+        return await SendRequest<Unit>(
+            () => httpClient.DeleteAsync(
+                url,
+                cancellationToken: cancellationToken),
+            cancellationToken);
     }
 
-    public async Task<Either<ActionResult, Unit>> DeleteDataSetVersion(
-        Guid dataSetVersionId,
+    private async Task<Either<ActionResult, TResponse>> SendRequest<TResponse>(
```

Comment:
I was speaking to Ben about this on one of my previous PR's. We were torn which way to go but in the end we decided not to break the handling of the HTTP response out into a common method - because it felt like each public method should be responsible for handling it's HTTP response individually, as the way it handles it **may** differ from endpoint to endpoint. For example, the `DeleteDataSetVersion` endpoint previously handled `NotFound` status codes; whereas now that case would fall into a `500`.

I'm still torn which way I'd go on it personally. It does save a lot of code, that is mostly just reused. But at the same time it feels like each endpoint should be handled individually as there isn't technically anything to say they won't be different. They're just the same now by coincidence (aside from `DeleteDataSetVersion` like I mentioned)